src/components/ReportActionItemIOUPreview.js

```diff
@@ -72,54 +85,69 @@ const ReportActionItemIOUPreview = ({
     const sessionEmail = lodashGet(session, 'email', null);
 
     // Pay button should only be visible to the manager of the report.
-    const isCurrentUserManager = iou.managerEmail === sessionEmail;
+    const isCurrentUserManager = iouReport.managerEmail === sessionEmail;
+    const reportIsLoading = _.isEmpty(iouReport);
+
+    if (reportIsLoading) {
+        fetchIOUReportByID(iouReportID, chatReportID);
+    }
 
     const managerName = lodashGet(
         personalDetails,
-        [iou.managerEmail, 'displayName'],
-        iou.managerEmail ? Str.removeSMSDomain(iou.managerEmail) : '',
+        [iouReport.managerEmail, 'displayName'],
+        iouReport.managerEmail ? Str.removeSMSDomain(iouReport.managerEmail) : '',
     );
     const ownerName = lodashGet(
         personalDetails,
-        [iou.ownerEmail, 'displayName'],
-        iou.ownerEmail ? Str.removeSMSDomain(iou.ownerEmail) : '',
+        [iouReport.ownerEmail, 'displayName'],
+        iouReport.ownerEmail ? Str.removeSMSDomain(iouReport.ownerEmail) : '',
     );
-    const managerAvatar = lodashGet(personalDetails, [iou.managerEmail, 'avatar'], '');
-    const ownerAvatar = lodashGet(personalDetails, [iou.ownerEmail, 'avatar'], '');
-    const cachedTotal = iou.cachedTotal ? iou.cachedTotal.replace(/[()]/g, '') : '';
+    const managerAvatar = lodashGet(personalDetails, [iouReport.managerEmail, 'avatar'], '');
+    const ownerAvatar = lodashGet(personalDetails, [iouReport.ownerEmail, 'avatar'], '');
+    const cachedTotal = iouReport.cachedTotal ? iouReport.cachedTotal.replace(/[()]/g, '') : '';
 
     return (
         <View style={styles.iouPreviewBox}>
-            <View style={styles.flexRow}>
-                <View style={styles.flex1}>
-                    <Text style={styles.h1}>{cachedTotal}</Text>
-                    <Text style={styles.mt2}>
-                        {iou.hasOutstandingIOU
-                            ? translate('iou.owes', {manager: managerName, owner: ownerName})
-                            : translate('iou.paid', {manager: managerName, owner: ownerName})}
-                    </Text>
-                </View>
-                <View style={styles.iouPreviewBoxAvatar}>
-                    <MultipleAvatars
-                        avatarImageURLs={[managerAvatar, ownerAvatar]}
-                        secondAvatarStyle={[styles.secondAvatarInline]}
-                    />
+            {reportIsLoading ? <ActivityIndicator color={themeColors.text} /> : (
+                <View>
```

Comment:
I'm not sure about this, but is it possible to remove this View?

Reply:
I'll try it out!

Reply:
Unfortunately, removing it gives me an `Unexpected token, expected "," (134:16)` error, I think the adjacent `})`s are the root of the issue. Adding `<View>` between them resolves that

Reply:
Messed around with it a bit and kept getting misleading errors like `Unexpected token, expected "," (134:16)` - I think since we end up conditionally rendering two adjacent views it requires some sort of separator.

Reply:
Ah whoops. No worries, thanks for giving it a try anyway!